core/src/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -3220,7 +3220,8 @@ public String getDataXML( Object object ) throws IOException {
             // at all.
             //
             string = XMLHandler.addTagValue( "binary-string", (byte[]) object );
-            break;
+            xml.append( XMLHandler.openTag( XML_DATA_TAG ) ).append( string ).append( XMLHandler.closeTag( XML_DATA_TAG ) );
```

Comment:
So isn't this what the code originally did except using xml.append( XMLHandler.addTagValue( XML_DATA_TAG, string ) ); instead of openTag/closeTag ? What is the difference between what it did before and what your change has ?